soroban-env-host/src/host.rs

```diff
@@ -107,6 +119,7 @@ pub struct LedgerInfo {
 
 #[derive(Clone, Default)]
 pub(crate) struct HostImpl {
+    source: RefCell<Option<Source>>,
```

Comment:
Todo: Change the type to simply `Option<AccountId>`.
```suggestion
    source_account: RefCell<Option<AccountId>>,
```

Reply:
We should also remove the Option. An account should always be set. In testutils we can default it to an all zero address. Wdyt?